test/suites/ekuiper/main_test.go

```diff
@@ -24,51 +24,26 @@ const (
 var testSecretsInterface bool
 
 func TestMain(m *testing.M) {
-	// start clean
-	utils.SnapRemove(nil,
-		ekuiperSnap,
-		"edgexfoundry",
-		deviceVirtualSnap,
-		ascSnap,
-	)
-
-	log.Println("[SETUP]")
-	start := time.Now()
-
-	// install the ekuiper snap before edgexfoundry
-	// to catch build error sooner and stop
-	if utils.LocalSnap() {
-		utils.SnapInstallFromFile(nil, utils.LocalSnapPath)
-	} else {
-		utils.SnapInstallFromStore(nil, ekuiperSnap, utils.ServiceChannel)
+	teardown, err := setupServiceTest(ekuiperSnap)
+	if err != nil {
+		log.Fatalf("Failed to setup tests: %s", err)
 	}
-	utils.SnapInstallFromStore(nil, "edgexfoundry", utils.PlatformChannel)
-	utils.SnapInstallFromStore(nil, deviceVirtualSnap, "latest/edge")
-	utils.SnapInstallFromStore(nil, ascSnap, "latest/edge")
 
 	// set profile to rules engine
 	utils.SnapSet(nil, ascSnap, "profile", "rules-engine")
 
-	// make sure all services are online before starting the tests
-	utils.WaitPlatformOnline(nil)
-
-	// for local build, the interface isn't auto-connected.
-	// connect manually
-	if utils.LocalSnap() {
-		utils.SnapConnect(nil,
-			"edgexfoundry:edgex-secretstore-token",
-			ekuiperSnap+":edgex-secretstore-token",
-		)
-	}
-
 	// security on (default)
 	testSecretsInterface = true
-	exitCode := m.Run()
-	if exitCode != 0 {
-		goto TEARDOWN
+
+	code := m.Run()
+	if code != 0 {
+		teardown()
+		os.Exit(code)
 	}
 
 	// security off
+	testSecretsInterface = false
```

Comment:
I suggest printing a log message here to show we are repeating the tests with disabled security.